### PR TITLE
[TIMOB-19067][TIMOB-23492] Implement reverseGeocoder() and forwardGeocoder() in TitaniumKit

### PR DESCRIPTION
**Pull request by garymathews**
- Move ``Titanium.Geolocation.reverseGeocoder()`` and ``Titanium.Geolocation.forwardGeocoder()`` into ``TitaniumKit``
- Use actual ``aguid`` ``mid`` and ``sid`` instead of placeholder
- Fixed parity issues in both ``reverseGeocoder()`` and ``forwardGeocoder()`` callbacks
- Improved test cases

###### TEST CASE
```Javascript
Ti.Geolocation.reverseGeocoder(40.7484, -73.9857, function (response) {
    Ti.API.info('reverseGeocoder: ' + JSON.stringify(response, null, ' '));
});
Ti.Geolocation.forwardGeocoder('Empire State Building, New York', function (response) {
    Ti.API.info('forwardGeocoder: ' + JSON.stringify(response, null, ' '));
});
```
```
[INFO] reverseGeocoder: {
 "success": true,
 "code": 0,
 "places": [
  {
   "country": "United States of America",
   "country_code": "US",
   "zipcode": "10018",
   "city": "New York City",
   "street": "5th Avenue",
   "state": "New York",
   "address": "Empire State Building, 350 5th Avenue, New York City, NY 10018, United States of America",
   "longitude": -73.9856545,
   "latitude": 40.7484284
  }
 ]
}
[INFO] forwardGeocoder: {
 "accuracy": "1",
 "address": "Empire State Building, New York",
 "latitude": 40.702289,
 "longitude": -74.014248,
 "success": true,
 "code": 0
}
```

[TIMOB-19067](https://jira.appcelerator.org/browse/TIMOB-19067)
[TIMOB-23492](https://jira.appcelerator.org/browse/TIMOB-23492)

### CODE REVIEW

File: Examples/NMocha/src/Assets/ti.geolocation.test.js [hunk 1]
```diff
@@ -159,18 +159,41 @@ describe('Titanium.Geolocation', function () {
...
-			should(data.longitude).be.eql(-122.0512682);
+		    try {
+		        should(data).have.property('success').which.is.a.Boolean;
+		        should(data.success).be.eql(true);
+		        should(data).have.property('code').which.is.a.Number;
+		        should(data.code).be.eql(0);
+		        should(data.latitude).be.eql(37.3883645);
+		        should(data.longitude).be.eql(-122.0512682);
+		    } catch (err) {
+		        finish();
```
Comment (infosia) [inline Examples/NMocha/src/Assets/ti.geolocation.test.js]: `finish()` only when exception is fired? why?

**Comment by infosia** [comment]
I'm not quite sure why unit tests `finish()`es only when it catches the exception. other than that, looks good to me 👍

**Comment by garymathews** [comment]
Oops nice catch!

**Comment by sgtcoolguy** [comment]
@garymathews Looks like the test is failing because of some floating point value comparisons. May want to borrow the updated tests for these methods from our common suite: https://github.com/appcelerator/titanium-mobile-mocha-suite/blob/master/Resources/ti.geolocation.test.js

**Comment by garymathews** [comment]
test this please

**Comment by garymathews** [comment]
build win 10

**Comment by garymathews** [comment]
build win 10

**Comment by infosia** [comment]
@garymathews, please update your branches with the base branch so that I can merge it. (It may be a pain because all of your branch goes out-of-date everytime base branch is updated...)

**Comment by sgtcoolguy** [comment]
build win 10

**Comment by sgtcoolguy** [comment]
@infosia I made you an Admin of the repo. As an admin, you have the ability to merge PRs if the branch is out of date with the base branch but there are no merge conflicts. Then you don't need to force each other to keep merging/rebasing as PRs get merged...

**Comment by infosia** [comment]
Awesome thanks @sgtcoolguy !

**Pull request merged**